### PR TITLE
Advise to use -XX:+ExitOnOutOfMemoryError in docs

### PR DESCRIPTION
**Pull request by findepi**
The flag is available since JDK 8u92. This is 9 months old already, so
it's safe to use.

### CODE REVIEW

File: presto-docs/src/main/sphinx/installation/deployment.rst [hunk 1]
```diff
@@ -80,7 +80,7 @@ The following provides a good starting point for creating ``etc/jvm.config``:
     -XX:+UseGCOverheadLimit
     -XX:+ExplicitGCInvokesConcurrent
     -XX:+HeapDumpOnOutOfMemoryError
-    -XX:OnOutOfMemoryError=kill -9 %p
```
Comment (electrum) [inline presto-docs/src/main/sphinx/installation/deployment.rst]: We should also remove the above reference to `OnOutOfMemoryError`.

File: presto-product-tests/conf/presto/etc/jvm.config [hunk 2]
```diff
@@ -13,7 +13,7 @@
 -XX:+CMSClassUnloadingEnabled
 -XX:+AggressiveOpts
 -XX:+HeapDumpOnOutOfMemoryError
--XX:OnOutOfMemoryError=kill -9 %p
+-XX:+ExitOnOutOfMemoryError
```
Comment (electrum) [inline presto-product-tests/conf/presto/etc/jvm.config]: This change means we require 8u92 for development, so we should also update the README and minimum version in the root POM.

File: presto-server-rpm/src/main/resources/dist/config/jvm.config [hunk 3]
```diff
@@ -5,5 +5,5 @@
 -XX:+ExplicitGCInvokesConcurrent
 -XX:+HeapDumpOnOutOfMemoryError
 -XX:+UseGCOverheadLimit
--XX:OnOutOfMemoryError=kill -9 %p
+-XX:+ExitOnOutOfMemoryError
```
Comment (electrum) [inline presto-server-rpm/src/main/resources/dist/config/jvm.config]: This change prevents the RPM installed version from running, so we should also change `preinstall` to require that version.

**Comment by findepi** [comment]
> Should we update the minimum required version everywhere? `PrestoSystemRequirements` also has a check for the JVM version.

I changed that too for consistency, but I admit this could be treated independently.

**Comment by findepi** [comment]
I applied comments. Are any further changes requested?

**Comment by findepi** [comment]
This is ready for re-review @electrum

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**